sources/Context.cpp

```diff
@@ -17,11 +17,26 @@
 
 #include "Context.h"
 
+#include "Log.h"
+
 #include <tuple>
 
 namespace dfl {
-Context::Context(const std::string& networkFilepath) : networkManager_(networkFilepath) {
-  networkManager_.onNode(std::bind(&Context::processSlackNode, this, std::placeholders::_1));
+Context::Context(const std::string& networkFilepath, const std::string& configFilepath) :
+    networkManager_(networkFilepath),
+    config_(configFilepath),
+    slack_node_{} {
+  auto automatic_slack_node = networkManager_.getSlackNode();
+  if (automatic_slack_node.is_initialized() && !config_.isAutomaticSlackBusOn()) {
+    slack_node_ = *automatic_slack_node;
+  } else {
+    // automatic slack node not given: we must compute it ourselves
+    if (!automatic_slack_node.is_initialized() && config_.isAutomaticSlackBusOn()) {
+      // TODO(lecourtoisflo) use warning without restriction for language
+      LOG(warn) << "Automatic slack node requested but not present in network input file" << LOG_ENDL;
```

Comment:
You should warn the user when Automatic=false and Found=false (in that case you warn and you calculate the slack node automatically)
If Automatic=true then calculate (whatever if Found or not)
If Automatic=false and Found=true, pick the slack found in the iidm

Reply:
OK